pam/internal/pam_test/pam-client-dummy_test.go

```diff
@@ -0,0 +1,1187 @@
+package pam_test
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/brokers"
+)
+
+var errTest = errors.New("an error")
+var privateKey *rsa.PrivateKey
+
+func TestAvailableBrokers(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		client authd.PAMClient
+
+		wantRet   *authd.ABResponse
+		wantError error
+	}{
+		"With empty options": {
+			client:  NewDummyClient(nil),
```

Comment:
Well, we do in the test of the client itself, but it should be discouraged otherwise I think, since we assume that a client has always a key set.

But also I think it would have made init func more complex. However, let me know if that's something you really want me to change and I'll address it.